File: python3/easy/mime-type.py
n, q = int(input()), int(input())
ext_to_mt = {ext.lower(): mt for ext, mt in [input().split() for _ in range(n)]}

for _ in range(q):
    fname = input().split(".")
    print(ext_to_mt.get(fname[-1].lower() if len(fname) > 1 else None, "UNKNOWN"))

File: javascript/easy/mime-type.js
[N, Q] = [parseInt(readline()), parseInt(readline())];

var extToMt = [...Array(N)]
    .map(n => readline())
    .reduce((r, e) => {
        r[e.split(' ')[0].toLowerCase()] = e.split(' ')[1];
        return r;
    }, {});

[...Array(Q)]
    .map(n => readline())
    .map((e) => {
        var ext = e.indexOf('.') !== -1 ? e.split('.').slice(-1).pop().toLowerCase() : null;
        print(ext in extToMt ? extToMt[ext] : 'UNKNOWN');
    });
